FreeRTOS_Sockets.c

```diff
@@ -386,8 +386,31 @@ Socket_t FreeRTOS_socket( BaseType_t xDomain,
     size_t uxSocketSize = 1;
     EventGroupHandle_t xEventGroup;
     Socket_t xReturn;
+    BaseType_t xProtocolOrDefault;
 
-    if( prvDetermineSocketSize( xDomain, xType, xProtocol, &uxSocketSize ) == pdFAIL )
+    /* A protocol of 0 indicates to the socket layer that it should pick a
+     * sensible default protocol based off the given socket type. If we can't,
+     * prvDetermineSocketSize will catch it as an invalid type/protocol combo.
+     */
+    if( xProtocol == 0 )
+    {
+        switch( xType )
+        {
+            case FREERTOS_SOCK_DGRAM:
+                xProtocolOrDefault = FREERTOS_IPPROTO_UDP;
+                break;
+
+            case FREERTOS_SOCK_STREAM:
+                xProtocolOrDefault = FREERTOS_IPPROTO_TCP;
+                break;
+        }
```

Comment:
Oops, if `xType` is undefined, `xProtocolOrDefault` will also become undefined.
Maybe just initialise it at the declaration?
~~~c
BaseType_t xProtocolOrDefault = xProtocol;
~~~
and forget about the `else`?

And a minor remark about the name `xProtocolOrDefault`: actually the parameter can be either 0 or denote a protocol. The variable `xProtocolOrDefault` should never be zero.

In other functions we solved the MISRA rule by introducing a new name follow by Copy, such as:
~~~c
    BaseType_t xProtocolCpy = xProtocol;
~~~

Reply:
I see your point. I addressed the issue.

Reply:
Thanks @htibosch for catching this and thank you @Zalasus for fixing the issue.